subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -3165,37 +3183,118 @@ static inline void event_phy_upd_ind_prep(struct ll_conn *conn,
 		/* apply new phy */
 		old_tx = lll->phy_tx;
 		old_rx = lll->phy_rx;
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+		u16_t eff_tx_time = lll->max_tx_time;
+		u16_t eff_rx_time = lll->max_rx_time;
+#endif /* CONFIG_BT_CTLR_DATA_LENGTH */
+
 		if (conn->llcp.phy_upd_ind.tx) {
 			lll->phy_tx = conn->llcp.phy_upd_ind.tx;
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+			u16_t tx_time = PKT_US(lll->max_tx_octets, lll->phy_tx);
+			if (tx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
+				eff_tx_time = MIN(tx_time,
+						  conn->default_tx_time);
+#if defined(CONFIG_BT_CTLR_PHY_CODED)
+				eff_tx_time = MAX(eff_tx_time,
+						  PKT_US(PDU_DC_PAYLOAD_SIZE_MIN,
+							 lll->phy_tx));
+#endif /* CONFIG_BT_CTLR_PHY_CODED */
+			} else {
+				eff_tx_time = PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0);
+			}
+#endif /* CONFIG_BT_CTLR_DATA_LENGTH */
```

Comment:
I would refactor this to (not even compiled, so to be taken with a grain of salt):
```
#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
			eff_tx_time = ull_conn_calc_eff_time(lll->max_tx_octets,
							lll->phy_tx,
							conn->default_tx_time);
#endif 
...
static u16_t ull_conn_calc_eff_time(max_octets, phy, default_time)
{
	u16_t eff_time;
	u16_t time = PKT_US(max_octets, phy);

	if (time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
		eff_time = MIN(time, default_time);
#if defined(CONFIG_BT_CTLR_PHY_CODED)
		eff_time = MAX(eff_time,
				PKT_US(PDU_DC_PAYLOAD_SIZE_MIN,	phy));
#endif /* CONFIG_BT_CTLR_PHY_CODED */
	} else {
		eff_time = PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0);
	}

	return eff_time;
}
```

ull_conn_calc_eff_time() could then be reused for the RX calculation below.